test/src/main/java/org/apache/accumulo/test/ScanServerMetadataEntriesIT.java

```diff
@@ -258,41 +244,43 @@ public void testGcRunScanServerReferences() throws Exception {
           String file = row.substring(ScanServerFileReferenceSection.getRowPrefix().length());
           metadataScanFileRefs.add(file);
         });
-        assertEquals(3, metadataScanFileRefs.size());
+        assertEquals(fileCount, metadataScanFileRefs.size());
 
-        assertEquals(3, ctx.getAmple().getScanServerFileReferences().count());
+        assertEquals(fileCount, ctx.getAmple().getScanServerFileReferences().count());
 
         List<Reference> refs = gc.getReferences().collect(Collectors.toList());
-        refs.forEach(ref -> LoggerFactory.getLogger(ScanServerMetadataEntriesIT.class)
-            .info("REF: {}", ref.getMetadataEntry()));
-        assertTrue(refs.size() > 6);
-        List<Reference> tableRefs = new ArrayList<>();
-        refs.forEach(r -> {
-          if (r.getTableId().equals(tid) && !r.isDirectory()) {
-            assertTrue(metadataScanFileRefs.contains(r.getMetadataEntry()));
-            tableRefs.add(r);
-          }
-        });
-        LoggerFactory.getLogger(ScanServerMetadataEntriesIT.class).info("{}", tableRefs);
+        refs.forEach(ref -> log.info("REF: {}", ref.getMetadataEntry()));
+        assertTrue(refs.size() > fileCount + 3);
+        List<Reference> tableRefs =
+            refs.stream().filter(r -> r.getTableId().equals(tid) && !r.isDirectory())
+                .peek(r -> assertTrue(metadataScanFileRefs.contains(r.getMetadataEntry())))
+                .collect(Collectors.toList());
+        log.info("Reference List:{}", tableRefs);
         // There should be 6 references here. 3 for the table file entries, and 3 for the scan
         // server references
-        assertEquals(6, tableRefs.size());
+        assertEquals(fileCount + 3, tableRefs.size());
```

Comment:
There may be some places in this test where I am miss-using this variable. It might be a good idea to add another variable for scan server references (which is also `int 3`)

Reply:
fileCount * 2 ?